addons/web/static/src/core/domain.js

```diff
@@ -135,8 +135,17 @@ export class Domain {
             /** @type {AST} */
             return new Domain(descr.toString());
         } else {
-            const rawAST = typeof descr === "string" ? parseExpr(descr) : toAST(descr);
-            this.ast = normalizeDomainAST(rawAST);
+            try {
+                const rawAST = typeof descr === "string" ? parseExpr(descr) : toAST(descr);
+                this.ast = normalizeDomainAST(rawAST);
```

Comment:
can this line crash with a random error? You don't need it in the try if it already throw the correct class of errors

Reply:
Right. Good remark. Thank you